scripts/generate_integration_test.py

```diff
@@ -0,0 +1,133 @@
+from typing import TextIO
+from urllib.parse import parse_qs, urlparse
+
+from playwright.sync_api import Playwright, Request, sync_playwright
+from structlog import get_logger
+
+logger = get_logger()
+
+
+previous_request_method = ""
+survey_start = True
+schema_name = ""
+
+LAUNCHER_ROOT_URL = "http://localhost:8000"
+RUNNER_ROOT_URL = "http://localhost:5000"
+
+
+# pylint: disable=global-variable-not-assigned
+def process_runner_request(request: Request) -> None:
+    global schema_name
+
+    with open(f"./scripts/{schema_name}.py", "a", encoding="utf-8") as file:
+        if request.method == "POST":
+            process_post(request, file)
+
+        elif request.method == "GET":
+            process_get(request, file)
+
+
+# pylint: disable=global-statement
+def process_post(request: Request, file: TextIO) -> None:
+    global previous_request_method
+
+    previous_request_method = "POST"
+
+    # Playwright Request.post_data comes formatted like a URL query string, so can be parsed
+    post_data = parse_qs(request.post_data)
+    del post_data["csrf_token"]
+
+    items: dict[str, str | list[str]] = {}
+    for answer_id, answer_values in post_data.items():
+        if len(answer_values) > 1:  # Handle multiple values in list
+            items[answer_id] = answer_values
+        elif len(answer_values) == 1:
+            items[answer_id] = answer_values[0]  # A single item in list
+
+    if items:
+        file.write(generate_method_request("post", items))
+    else:
+        # Empty post for no answers and non question pages
+        file.write(generate_method_request("post", ""))
+
+
+# pylint: disable=global-statement
+def process_get(request: Request, file: TextIO) -> None:
+    global previous_request_method
+    global survey_start
+
+    if (  # Logic to filter out the session and root questionnaire GET requests at the start
+        previous_request_method == "GET"
+        and "session?token" not in request.url
+        and request.url != f"{RUNNER_ROOT_URL}/questionnaire/"
+    ):
+        path = f'"{urlparse(request.url).path}"'
```

Comment:
I think this was so the self.post and self.get had the data in quotes. If we add the quotes in the `generate_method_request()` and data is `None`, it would end up with something like `self.post("")` instead of `self.post()`